wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -418,6 +419,43 @@ public static function generateQrCode($collectionCampId) {
     return TRUE;
   }
 
+  /**
+   * This hook is called after a db write on entities.
+   *
+   * @param string $op
+   *   The type of operation being performed.
+   * @param string $objectName
+   *   The name of the object.
+   * @param int $objectId
+   *   The unique identifier for the object.
+   * @param object $objectRef
+   *   The reference to the object.
+   */
+  public static function reGenerateCollectionCampQr(string $op, string $objectName, int $objectId, &$objectRef) {
+    // Check if the object name is 'Eck_Collection_Camp'.
+    if ($objectName !== 'Eck_Collection_Camp') {
+      return;
+    }
+
+    $collectionCampId = $objectRef->id ?? NULL;
```

Comment:
In the above the check can't we do this: `$objectName !== 'Eck_Collection_Camp' || !$objectRef->id`